pkg/bpf/map_linux.go

```diff
@@ -402,38 +397,32 @@ func GetMapInfo(pid int, fd int) (*MapInfo, error) {
 // *Warning*: Calling this function requires the caller to properly setup
 // the MapInfo.MapKey and MapInfo.MapValues fields as those structures are not
 // stored in the bpf map.
-func OpenMap(name string) (*Map, error) {
-	// Expand path if needed
-	if !path.IsAbs(name) {
-		name = MapPath(name)
+func OpenMap(pinPath string) (*Map, error) {
+	if !path.IsAbs(pinPath) {
+		return nil, fmt.Errorf("pinPath must be absolute: %s", pinPath)
 	}
 
-	fd, err := ObjGet(name)
+	em, err := ebpf.LoadPinnedMap(pinPath, nil)
 	if err != nil {
 		return nil, err
 	}
 
-	info, err := GetMapInfo(os.Getpid(), fd)
-	if err != nil {
-		return nil, err
-	}
-
-	if info.MapType == 0 {
-		return nil, fmt.Errorf("Unable to determine map type")
-	}
-
-	if info.KeySize == 0 {
-		return nil, fmt.Errorf("Unable to determine map key size")
-	}
-
 	m := &Map{
-		MapInfo: *info,
-		fd:      fd,
-		name:    path.Base(name),
-		path:    name,
+		MapInfo: MapInfo{
+			MapType:       MapType(em.Type()),
+			KeySize:       em.KeySize(),
+			ValueSize:     em.ValueSize(),
+			ReadValueSize: em.ValueSize(),
```

Comment:
Not sure whether this is actually used in the code base, but this value is going to be wrong for per CPU maps, no?

Reply:
Nope, fdinfo of percpu hashes shows the value size given during creation, not the total over all CPUs. Turns out `pkg/maps/metricsmap.metricsMap` used by `cilium bpf metrics list` is already implemented on top of `pkg/ebpf` using ebpf-go.

Given the code here replaces `GetMapInfo` behaviour 1:1, this is not going to break anything.